packages/fx-tunnel-relayer/src/Relayer.ts

```diff
@@ -0,0 +1,126 @@
+    this.logger.debug({
+      at: "Relayer#relayMessage",
+      message: "Found MessageSent events",
+      polygonEarliestBlockToQuery: this.polygonEarliestBlockToQuery,
+      eventCount: messageSentEvents.length,
+    });
+    // For any found events, grab its block number and check whether it has been checkpointed yet to the
+    // CheckpointManager on Ethereum.
+    if (messageSentEvents.length > 0) {
+      for (const e of messageSentEvents) {
+        await this._relayMessage(e);
+      }
+    } else {
+      this.logger.debug({
+        at: "Relayer#relayMessage",
+        message: "No MessageSent events emitted by OracleChildTunnel, exiting",
+      });
+      return;
+    }
+  }
+
+  // First check if the transaction hash corresponding to the MessageSent event has been checkpointed to Ethereum
+  // Mainnet yet. If it has, then derive a Polygon-specific proof for it and execute OracleRootTunnel.receiveMessage
+  // by passing in the proof as input.
+  async _relayMessage(messageEvent: EventData): Promise<void> {
+    const transactionHash = messageEvent.transactionHash;
+    this.logger.debug({
+      at: "Relayer#relayMessage",
+      message: "Deriving proof for transaction that emitted MessageSent",
+      transactionHash: transactionHash,
+      blockNumber: messageEvent.blockNumber,
+    });
+
+    // This method will fail if the Polygon transaction hash has not been checkpointed to Mainnet yet. Checkpoints
+    // happen roughly every hour.
+    let proof;
+    try {
+      // Proof construction logic copied from:
+      // - https://docs.matic.network/docs/develop/l1-l2-communication/state-transfer#state-transfer-from-matic-to-ethereum
+      proof = await this.maticPosClient.posRootChainManager.customPayload(
+        transactionHash,
+        POLYGON_MESSAGE_SENT_EVENT_SIG // SEND_MESSAGE_EVENT_SIG, do not change
+      );
+      if (!proof) throw new Error("Proof construction succeeded but returned undefined");
+    } catch (error) {
+      // If event block hasn't been checkpointed to Mainnet yet, then don't emit error level log.
+      let logLevel = "error";
+      if (error.message.includes("transaction has not been checkpointed")) logLevel = "debug";
+      this.logger[logLevel]({
+        at: "Relayer#relayMessage",
+        message: "Failed to derive proof for MessageSent transaction hash 📛",
+        messageEvent,
+        error,
+      });
+      return;
+    }
+
+    // Simulate and submit receiveMessage transaction with proof as input.
+    this.logger.debug({
+      at: "Relayer#relayMessage",
+      message: "Submitting proof",
+      proof: proof,
+      account: this.account,
+    });
+    try {
+      const { receipt } = await runTransaction({
+        web3: this.web3,
+        transaction: this.oracleRootTunnel.methods.receiveMessage(proof),
+        transactionConfig: { gasPrice: this.gasEstimator.getCurrentFastPrice(), from: this.account },
+        availableAccounts: 1,
```

Comment:
You should not need to define this. It defaults to 1 I think

Reply:
If I remove this I get this error:

```sh
src/Relayer.ts:106:48 - error TS2345: Argument of type '{ web3: Web3; transaction: any; transactionConfig: { gasPrice: any; from: string; }; }' is not assignable to parameter of type '{ web3: Web3; transaction: ContractSendMethod; transactionConfig: AugmentedSendOptions; availableAccounts: number; }'.
  Property 'availableAccounts' is missing in type '{ web3: Web3; transaction: any; transactionConfig: { gasPrice: any; from: string; }; }' but required in type '{ web3: Web3; transaction: ContractSendMethod; transactionConfig: AugmentedSendOptions; availableAccounts: number; }'.

106       const { receipt } = await runTransaction({
                                                   ~
107         web3: this.web3,
    ~~~~~~~~~~~~~~~~~~~~~~~~
... 
109         transactionConfig: { gasPrice: this.gasEstimator.getCurrentFastPrice(), from: this.account },
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
110       });
    ~~~~~~~

  ../common/dist/TransactionUtils.d.ts:23:5
    23     availableAccounts: number;
           ~~~~~~~~~~~~~~~~~
    'availableAccounts' is declared here.


Found 1 error.

```

Reply:
damn typescript. fine then

Reply:
Yeah, not sure there's a way to both define a default value AND make a param optional in typescript